packages/astro/src/vite-plugin-astro-server/base.ts

```diff
@@ -52,11 +52,13 @@ export function baseMiddleware(
 		const publicPath = new URL('.' + req.url, config.publicDir);
 		fs.stat(publicPath, (_err, stats) => {
 			if (stats) {
-				const expectedLocation = new URL('.' + url, devRootURL).pathname;
+				const publicDir = config.publicDir.pathname.replace(/\/$/, '').split('/').pop() + '/';
```

Comment:
I think there is a utility for slashes somewhere, where is it located @natemoo-re?

Reply:
Here's an example: https://github.com/withastro/astro/blob/main/packages%2Fastro%2Fsrc%2Fi18n%2Findex.ts#L1

Reply:
Like this? : ) https://github.com/withastro/astro/pull/9841/commits/a9afa2bd577f83af06eb717c5f4cd482aeedc628